### ISSUE TITLE
Mention the difference between `with TestClient(app) as client` and `client = TestClient(app)`

### ISSUE THREAD

**Issue by Kludex** [Kludex/starlette#1733]
Add a note on the `TestClient` section about the behavior difference between those two ways, i.e. lifespan events.

Considering the follow application:
```python
from starlette.applications import Starlette
from starlette.requests import Request
from starlette.responses import Response
from starlette.routing import Route
from starlette.testclient import TestClient


async def startup() -> None:
    print("Startup event!")

async def endpoint(request: Request) -> Response:
    return Response(status_code=204)

app = Starlette(routes=[Route("/", endpoint)], on_startup=[startup])
```

The following `TestClient` prints "Startup event!":
```python
with TestClient(app) as client:
    ...
```

But the following, doesn't:
```python
client = TestClient(app)
```

**Comment by adriangb** [comment]
Can we use the async context manager lifespans instead of startup/shutdown events?

Related to this: do we want to mention that the context manager just runs the lifespan, it doesn't "reset" the app instance, so you _can_ leak state between tests?

**Comment by Kludex** [comment]
> Can we use the async context manager lifespans instead of startup/shutdown events?

wdym?

> Related to this: do we want to mention that the context manager just runs the lifespan, it doesn't "reset" the app instance, so you can leak state between tests?

I don't think it's necessary, but... 🤷‍♂️

**Comment by adriangb** [comment]
```python
@contextlib.asynccontextmanager
async def lifespan(app):
    async with some_async_resource():
        yield
```

I thought this was the "recommended" way to do lifespans nowdays

**Comment by Kludex** [comment]
I've just noticed that we do have a note about the behavior above, but on the `Events` page: https://www.starlette.io/events/#running-event-handlers-in-tests . We can add a note and link this reference on the `TestClient` page.

**Comment by rednafi** [comment]
Yep. The event page already has a [section](https://www.starlette.io/events/#running-event-handlers-in-tests) that discusses it. I sent a PR with a note that links this section on the TestClient page.

**Comment by Kludex** [comment]
Closed by #1747
